### PR TITLE
Share code between retain(...) and release(...) implementations.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We can share the code in retain() and retain(...) and also in release() and release(...).

Modifications:

Share code.

Result:

Less duplicated code.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@Scottmitch @nmittler @trustin PTAL

**Comment by nmittler** [comment]
LGTM

File: common/src/main/java/io/netty/util/AbstractReferenceCounted.java [hunk 1]
```diff
@@ -51,27 +53,15 @@ protected final void setRefCnt(int refCnt) {
...
     @Override
     public ReferenceCounted retain(int increment) {
-        if (increment <= 0) {
-            throw new IllegalArgumentException("increment: " + increment + " (expected: > 0)");
-        }
+        return retain0(checkPositive(increment, "increment"));
+    }
 
+    private ReferenceCounted retain0(int increment) {
         for (;;) {
```
Comment (Scottmitch) [inline common/src/main/java/io/netty/util/AbstractReferenceCounted.java]: does it make sense to put this code in an internal static utility method and share it between the two classes?

Comment (normanmaurer) [inline common/src/main/java/io/netty/util/AbstractReferenceCounted.java]: @nmittler I was thinking about this but the `AtomicIntegerFieldUpdater` needs to defined in the class where it is used (as far as I remember). Let me give it a try again tho.

Comment (normanmaurer) [inline common/src/main/java/io/netty/util/AbstractReferenceCounted.java]: @Scottmitch this utility stuff feels ugly. let us just duplicate the code for now.

Comment (nmittler) [inline common/src/main/java/io/netty/util/AbstractReferenceCounted.java]: sgtm

**Comment by trustin** [comment]
LGTM

**Comment by Scottmitch** [comment]
lgtm

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (3103f0551cc0d16c7dda5bad385fc7d6a34bce6b) and 4.0 (a749b8aca84e2680fd22f285968ba3adac221f9f)

**Pull request closed without merging**